sdk/storage/azure-storage-data-movement/src/main/java/com/azure/storage/data/movement/PathScanner.java

```diff
@@ -0,0 +1,90 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.storage.data.movement;
+
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+import com.azure.core.util.logging.ClientLogger;
+import reactor.core.scheduler.Schedulers;
+
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.stream.BaseStream;
+
+/**
+ * Scanner for enumerating local files/folders at a specified location through publishers.
+ *
+ * TODO: Replace placeholder Javadoc
+ */
+public final class PathScanner {
+    private final ClientLogger logger = new ClientLogger(PathScanner.class);
+    private final String basePath;
+
+    /**
+     * Constructor for {@link PathScanner}.
+     *
+     * TODO: Replace placeholder Javadoc
+     *
+     * @param path The local path to be scanned, either relative to execution location or absolute.
+     * @throws IllegalArgumentException if the provided path does not point to anything.
+     */
+    public PathScanner(String path) {
+        try {
+            // Resolve the given path to an absolute path in case it isn't one already
+            Path pathObj = Paths.get(path);
+            basePath = pathObj.toAbsolutePath().toString();
+
+            // Check if the path exists; throw an error if there's nothing
+            // present at the given path
+            if (!(Files.isDirectory(pathObj) || Files.isRegularFile(pathObj))) {
+                throw new IllegalArgumentException(String.format("No item(s) located at the path '%s'.", basePath));
+            }
+        } catch (RuntimeException e) {
+            // Possible throws: nonexistent item, malformed path
+            throw logger.logExceptionAsError(e);
```

Comment:
Presumably this exception is going to bubble up and becomes caller responsibility to handle it. Therefore I don't think it's reasonable to assume it's an error - it should be warn at most.
I'd rather remove this whole try catch block alltogether - we don't know if we need it.

see 8 here https://stackify.com/best-practices-exceptions-java/

Reply:
So is it fine if I just have like, `throw logger.logWarning(new IllegalException...)` in the if, and remove the try-catch?